packages/ord-connect/src/components/SelectWalletModal/index.tsx

```diff
@@ -69,10 +69,10 @@ export function SelectWalletModal({
     [],
   );
 
+  // eslint-disable-next-line @typescript-eslint/no-use-before-define
+  const listener = async () => onConnectUnisatWallet();
+
   const onConnectUnisatWallet = async (readOnly?: boolean) => {
-    const listener = () => {
-      onConnectUnisatWallet();
-    };
     try {
```

Comment:
I haven't tried removing this part yet but I think it can be removed

I think the reason why it failed on first run was because the extension was not ready but this is already accounted for now using the check in the effect below.